### PR TITLE
Metrics server with capabilities to fetch response from elasticsearch via request Puid in feedback request

### PR DESCRIPTION
**Pull request by axsaucedo**
**What this PR does / why we need it**:

Introduces further capability from infrastructure metrics workingroup to enable "RESPONSE" to be fetched directly from the elasticsearch database. PR includes:
* Extra parameter for adserver to initialise elasticsearch client
* Custom metrics able to fetch RESPONSE if request PUID is provided
* Fixes on custom metrics server to work with param ndarrays
* Example showing the workflow using metrics server with feedback containing request Puid

[Jupyter notebook](https://github.com/axsaucedo/seldon-core/blob/9f45e48561c6b45bef53df3fbef7d048e4fe746b/examples/feedback/feedback-metrics-server/README.ipynb) and [README version](https://github.com/axsaucedo/seldon-core/blob/9f45e48561c6b45bef53df3fbef7d048e4fe746b/examples/feedback/feedback-metrics-server/README.md) of example.

Further discussion points:
* Currently it's required to pass request Puid as feedback Puid header, however this may be ambiguous if we want to eventually have unique feedback UIDs
* Current elasticsearch configuration is quite basic, but there could be some exploration to leverage the logic in the payload request logger to interact with ELK - this could involve merging the payload request logger with the CEServer as it's already quite modular.

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **axsaucedo**
You can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Nov  2 19:16:34 UTC 2020
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=2

**Comment by seldondev** [comment]
Mon Nov  2 19:16:47 UTC 2020
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=1

**Comment by seldondev** [comment]
Mon Nov  2 19:21:09 UTC 2020
The logs for [notebooks] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=3

**Comment by seldondev** [comment]
Tue Nov  3 14:01:52 UTC 2020
The logs for [pr-build] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=4

**Comment by seldondev** [comment]
Tue Nov  3 14:02:25 UTC 2020
The logs for [lint] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=5

**Comment by seldondev** [comment]
@axsaucedo: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
pr-build | 0e18e821e7351dd9d6d575dac862e80ee1684bcd | [link]() | `/test this`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Fri Nov  6 06:57:31 UTC 2020
The logs for [pr-build] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=6

File: components/alibi-detect-server/Dockerfile
Comment (cliveseldon) [inline components/alibi-detect-server/Dockerfile:2]: Shoud it not be 1.5.0-dev

**Review by cliveseldon — CHANGES_REQUESTED** [review]

File: examples/feedback/feedback-metrics-server/config/multiclass-deployment.yaml
Comment (cliveseldon) [inline examples/feedback/feedback-metrics-server/config/multiclass-deployment.yaml:24]: same - 1.5.0-dev
Might want to work out a way to have these take of the currebt version.txt?

File: components/alibi-detect-server/Dockerfile
Comment (axsaucedo) [reply components/alibi-detect-server/Dockerfile:2]: Oh wow good catch, yeah it should, I think i need to rebase actually... it seems this one is not covered so I shouldn't get any conflicts

Comment (axsaucedo) [reply components/alibi-detect-server/Dockerfile]: Ok this is now done 👍

**Comment by seldondev** [comment]
Fri Nov  6 13:55:26 UTC 2020
The logs for [pr-build] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=7

File: examples/feedback/feedback-metrics-server/config/multiclass-service.yaml
Comment (cliveseldon) [inline examples/feedback/feedback-metrics-server/config/multiclass-service.yaml:12]: missed this?

**Comment by seldondev** [comment]
Fri Nov  6 13:55:45 UTC 2020
The logs for [lint] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=8

**Comment by seldondev** [comment]
Fri Nov  6 14:02:43 UTC 2020
The logs for [pr-build] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=9

**Comment by seldondev** [comment]
Fri Nov  6 14:03:36 UTC 2020
The logs for [lint] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=10

**Comment by seldondev** [comment]
Fri Nov  6 14:07:21 UTC 2020
The logs for [pr-build] [11] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/11.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=11

**Comment by seldondev** [comment]
Fri Nov  6 14:07:45 UTC 2020
The logs for [lint] [12] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/12.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=12

**Comment by seldondev** [comment]
Fri Nov  6 15:08:28 UTC 2020
The logs for [lint] [14] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/14.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=14

**Comment by seldondev** [comment]
Fri Nov  6 15:08:41 UTC 2020
The logs for [pr-build] [13] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/13.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=13

**Comment by seldondev** [comment]
Fri Nov  6 15:09:21 UTC 2020
The logs for [pr-build] [15] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/15.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=15

**Comment by seldondev** [comment]
Fri Nov  6 15:09:22 UTC 2020
The logs for [lint] [16] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/16.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=16

**Comment by seldondev** [comment]
Fri Nov  6 15:10:04 UTC 2020
The logs for [pr-build] [17] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/17.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=17

**Comment by seldondev** [comment]
Fri Nov  6 15:11:16 UTC 2020
The logs for [lint] [18] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/18.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=18

**Comment by seldondev** [comment]
Fri Nov  6 15:28:08 UTC 2020
The logs for [lint] [20] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/20.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=20

**Comment by seldondev** [comment]
Fri Nov  6 15:29:11 UTC 2020
The logs for [pr-build] [19] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/19.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=19

**Comment by axsaucedo** [comment]
Merging as discussed in sc meeting
/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[axsaucedo](https://github.com/SeldonIO/seldon-core/pull/2604#issuecomment-725982397 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com#issuecomment-725982397/SeldonIO/seldon-core/blob/master/OWNERS)~~ [axsaucedo]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Thu Nov 12 10:17:28 UTC 2020
The logs for [lint] [22] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/22.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=22

**Comment by seldondev** [comment]
Thu Nov 12 10:17:28 UTC 2020
The logs for [pr-build] [21] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/21.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=21

**Comment by seldondev** [comment]
Thu Nov 12 10:35:35 UTC 2020
The logs for [lint] [24] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/24.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=24

**Comment by seldondev** [comment]
Thu Nov 12 10:38:23 UTC 2020
The logs for [pr-build] [23] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2604/23.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2604 --build=23

**Pull request merged**
